app/lib/account/backend.dart

```diff
@@ -290,79 +291,111 @@ class AccountBackend {
   }
 
   Future<User> _lookupOrCreateUserByOauthUserId(AuthResult auth) async {
+    ArgumentError.checkNotNull(auth, 'auth');
     if (auth.oauthUserId == null) {
       throw StateError('Authenticated user ${auth.email} without userId.');
     }
-    final mappingKey = _db.emptyKey.append(OAuthUserID, id: auth.oauthUserId);
-
-    final user = await retry(() async {
-      // Check existing mapping.
-      final mapping = (await _db.lookup<OAuthUserID>([mappingKey])).single;
-      if (mapping != null) {
-        final user = (await _db.lookup<User>([mapping.userIdKey])).single;
-        // TODO: we should probably have some kind of consistency mitigation
-        if (user == null) {
-          throw StateError('Incomplete OAuth userId mapping: '
-              'missing User (`${mapping.userId}`) referenced by `${mapping.id}`.');
-        }
-        return user;
-      }
 
-      // Check pre-migrated User with existing email.
-      final usersWithEmail = await (_db.query<User>()
-            ..filter('email =', auth.email))
-          .run()
-          .toList();
-      if (usersWithEmail.length == 1 &&
-          usersWithEmail.single.oauthUserId == null &&
-          !usersWithEmail.single.isDeleted) {
-        // We've found a single pre-migrated, non-deleted User with empty
-        // `oauthUserId` field: need to create OAuthUserID for it.
-        final updatedUser = await _db.withTransaction<User>((tx) async {
-          final user =
-              (await tx.lookup<User>([usersWithEmail.single.key])).single;
-          final newMapping = OAuthUserID()
-            ..parentKey = _db.emptyKey
-            ..id = auth.oauthUserId
-            ..userIdKey = user.key;
-          user.oauthUserId = auth.oauthUserId;
-          tx.queueMutations(inserts: [user, newMapping]);
-          await tx.commit();
-          return user;
-        });
-        return updatedUser;
+    final emptyKey = _db.emptyKey;
+
+    // Attempt to lookup the user, the common case is that the user exists.
+    // If the user exists, it's always cheaper to lookup the user outside a
+    // transaction.
+    // TODO: This should be cached.
+    {
```

Comment:
this code block / indent is really weird, remove?

Reply:
Moved the first part to separate helper method.